### ISSUE TITLE
[BUG] Exception when PYTHONOPTIMIZE=2

### ISSUE THREAD

**Issue by lmeyerov** [rapidsai/cudf#4930]
**Describe the bug**

Running with `PYTHONOPTIMIZE=2` throws a runtime exception

Level 2 strips doc strings, so looks like the decorators need to be disabled when off (`not __debug__`? Something like that)

**Steps/Code to reproduce bug**

Run with `PYTHONOPTIMIZE=2` with a column import

```
from cudf import Series
Series([1, 2, 3])
```
...
```
PYTHONOPTIMIZE=2 python ...
```

**Expected behavior**

Non-crashing

** Actual behavior**

```
forge-etl-python_1     | 2020-04-17T16:19:10.022648830Z   File "/opt/graphistry/apps/forge/etl-server-python/server/main.py", line 19, in init
forge-etl-python_1     | 2020-04-17T16:19:10.022656832Z     from server.util.cudf import test_and_warm_cudf
forge-etl-python_1     | 2020-04-17T16:19:10.022664609Z   File "/opt/graphistry/apps/forge/etl-server-python/server/util/cudf.py", line 3, in <module>
forge-etl-python_1     | 2020-04-17T16:19:10.022673618Z     import cudf, gc, json, os, sys, time
forge-etl-python_1     | 2020-04-17T16:19:10.022684090Z   File "/conda/envs/rapids/lib/python3.7/site-packages/cudf/__init__.py", line 7, in <module>
forge-etl-python_1     | 2020-04-17T16:19:10.022694004Z     from cudf import core, datasets
forge-etl-python_1     | 2020-04-17T16:19:10.022701674Z   File "/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/__init__.py", line 3, in <module>
forge-etl-python_1     | 2020-04-17T16:19:10.022710834Z     from cudf.core import buffer, column
forge-etl-python_1     | 2020-04-17T16:19:10.022720433Z   File "/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/column/__init__.py", line 1, in <module>
forge-etl-python_1     | 2020-04-17T16:19:10.022731140Z     from cudf.core.column.categorical import CategoricalColumn  # noqa: F401
forge-etl-python_1     | 2020-04-17T16:19:10.022739113Z   File "/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/column/categorical.py", line 14, in <module>
forge-etl-python_1     | 2020-04-17T16:19:10.022747865Z     from cudf.core.column import column
forge-etl-python_1     | 2020-04-17T16:19:10.022756563Z   File "/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/column/column.py", line 41, in <module>
forge-etl-python_1     | 2020-04-17T16:19:10.022766549Z     class ColumnBase(Column):
forge-etl-python_1     | 2020-04-17T16:19:10.022774675Z   File "/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/column/column.py", line 692, in ColumnBase
forge-etl-python_1     | 2020-04-17T16:19:10.022783345Z     @ioutils.doc_to_dlpack()
forge-etl-python_1     | 2020-04-17T16:19:10.022791967Z   File "/conda/envs/rapids/lib/python3.7/site-packages/cudf/utils/docutils.py", line 30, in outer
forge-etl-python_1     | 2020-04-17T16:19:10.022801812Z     formatsiter = string.Formatter().parse(fn.__doc__)
forge-etl-python_1     | 2020-04-17T16:19:10.022811276Z   File "/conda/envs/rapids/lib/python3.7/string.py", line 284, in parse
forge-etl-python_1     | 2020-04-17T16:19:10.022833384Z     return _string.formatter_parser(format_string)
```

**Environment overview (please complete the following information)**

RAPIDS 0.13 (conda) in docker

**Comment by lmeyerov** [comment]
Was this fixed?

**Comment by kkraus14** [comment]
Yes, this was fixed in #7434.
